python/rola/utils/gateway.py

```diff
@@ -0,0 +1,49 @@
+import requests
+from radix_engine_toolkit import PublicKeyHash
+from abc import ABC, abstractmethod
+
+from typing import List
+
+from rola.exceptions.gateway import EntityNotFound
+
+
+class MetadataProvider(ABC):
+    @abstractmethod
+    def entity_owner(self, address: str) -> List[PublicKeyHash]:
+        pass
+
+
+class GatewayMetadataProvider:
+    base_url: str
+
+    def __init__(self, base_url: str = "https://mainnet.radixdlt.com") -> None:
+        self.base_url = base_url
+
+    @classmethod
+    def for_mainnet(cls) -> "GatewayMetadataProvider":
+        return cls("https://mainnet.radixdlt.com")
+
+    @classmethod
+    def for_stokenet(cls) -> "GatewayMetadataProvider":
+        return cls("https://stokenet.radixdlt.com")
+
+    def entity_owner(self, address: str) -> PublicKeyHash.ED25519:
+        headers = {"accept": "application/json"}
+        body = {"addresses": [address]}
+        response = requests.post(
+            url=f"{self.base_url}/state/entity/details", headers=headers, json=body
+        )
+
+        if response.status_code != 200:
+            response.raise_for_status()
+
+        items = response.json().get("items", [])
+        if not items:
+            raise EntityNotFound()
+
+        metadata_items = items[0].get("metadata", {}).get("items", [])
+        owner_keys = [item for item in metadata_items if item["key"] == "owner_keys"]
+        if len(owner_keys) == 1:
+            return PublicKeyHash.ED25519(owner_keys[0]["value"]["raw_hex"])
```

Comment:
Few things about this:

1. The `owner_keys` is a `list[PublicKeyHash]` and not just a single public key hash.
2. The `owner_keys` might not be on curve ed25519 so we must return a generic `list[PublicKeyHash]`
3. The `raw_hex` is SBOR encoded `Vec<PublicKeyHash>` and not just an individual public key hash